src/pages/TeamOverview/TeamOverview.tsx

```diff
@@ -35,42 +36,71 @@ const getTeamLeadCard = (teamLead: UserDataType): JSX.Element => {
     return <Card id={id} data-testid='team-lead' columns={columns} url={`/user/${id}`} navigationProps={teamLead} />;
 };
 
-interface PageState {
-    teamLead?: UserDataType;
-    teamMembers?: UserDataType[];
-}
-
 const TeamOverview = (): JSX.Element => {
     const location: Location<TeamsType> = useLocation();
     const {teamId} = useParams();
-    const [pageData, setPageData] = React.useState<PageState>({});
+    const [allTeamMembers, setAllTeamMembers] = React.useState<UserDataType[]>([]);
+    const [filteredTeamMembers, setFilteredTeamMembers] = React.useState<UserDataType[]>([]);
     const [isLoading, setIsLoading] = React.useState<boolean>(true);
+    const [searchError, setSearchError] = React.useState<boolean>(false);
+    const [leadId, setLeadId] = React.useState<string>('');
+    const [showTeamLead, setShowTeamLead] = React.useState<boolean>(true);
+
+    const noMembersMessage = 'No team member exists with this search.';
 
     React.useEffect(() => {
         const getTeamUsers = async () => {
             const {teamLeadId, teamMemberIds = []} = await getTeamOverview(teamId);
             const teamLead = await getUserData(teamLeadId);
+            setLeadId(teamLeadId);
 
             const teamMembers = [];
             for(const teamMemberId of teamMemberIds) {
                 const data = await getUserData(teamMemberId);
                 teamMembers.push(data);
             }
-            setPageData({
-                teamLead,
-                teamMembers,
-            });
+            setFilteredTeamMembers([teamLead, ...teamMembers]);
+            setAllTeamMembers([teamLead, ...teamMembers]);
             setIsLoading(false);
         };
 
         getTeamUsers();
     }, [teamId]);
 
+    React.useEffect(() => {
```

Comment:
Check if the `teamLead` is in the filteredArrays, if not, hide it from the page (it's always displayed from the original array, to avoid problems with ordering)